### PR TITLE
Add Support for `modulus` operation in substrait

### PR DESCRIPTION
**Pull request by LatrecheYasser**
## Which issue does this PR close?

This PR replaces adds the operation name `modulus` which is the standard [Substrait operation name](https://substrait.io/extensions/functions_arithmetic/#modulus) for `mod`

### CODE REVIEW

File: datafusion/substrait/src/logical_plan/consumer.rs [hunk 1]
```diff
@@ -108,6 +108,7 @@ pub fn name_to_op(name: &str) -> Option<Operator> {
         "multiply" => Some(Operator::Multiply),
         "divide" => Some(Operator::Divide),
         "mod" => Some(Operator::Modulo),
```
Comment (LatrecheYasser) [inline datafusion/substrait/src/logical_plan/consumer.rs:110]: because `mod` is the same as `modulus` we need to mark it as deprecated somehow, but how?

**Review by alamb — COMMENTED** [review]
Thank you @LatrecheYasser 

cc @Blizzara  @westonpace and @vbarua

File: datafusion/substrait/src/logical_plan/consumer.rs (hunk 1 above)
Comment (vbarua) [reply datafusion/substrait/src/logical_plan/consumer.rs:110]: I'm not sure how to produce a deprecation warning for this programatically.

However, we can provide an easy migration path if we also update the producer code to use `modulus` so we can release a version of DataFusion that consumes `mod` and `modulus`, and produces only `modulus`. Then the version after than we remove `mod` and announce it as a breaking change in the release notes.

**Review by alamb — APPROVED** [review]
This change seems good to me -- thank you @LatrecheYasser  and @vbarua 

My only concern is that it isn't tested (and no tests needed to be updated either). 

But maybe that is ok, as this PR doesn't improve / make the testing situation worse

**Comment by Blizzara** [comment]
LGTM. Tests don’t need updating as we mainly test the roundtrips (I didn’t check if we test even that for this specific function, but even if we did, it still would just keep passing.) I dunno if it makes sense to manually test all functions, but I think nowdays the substrait-rs might include the Substrait default function definition yamls so we could maybe do something to compare the produced functions against them (which would have caught the ”mod” before)..

**Comment by LatrecheYasser** [comment]
cc @alamb  @Blizzara 
I've added a simple roundtrip test case for modulus in  766dacae7f47f165bfe73f8ea5a086f99a8f19dc

**Comment by alamb** [comment]
Thank you everyone!

**Pull request merged**